models/convert/cadence_test.go

```diff
@@ -75,7 +104,7 @@ func TestParseCadenceArgument(t *testing.T) {
 		gotArg, err := ParseCadenceArgument(vector.param)
 		vector.checkErr(t, err)
 
-		if err != nil {
+		if err == nil {
 			assert.Equal(t, vector.wantArg, gotArg)
 		}
```

Comment:
No, I believe that's actually just _wrong_. The Go specification is imo clear that no assumptions should be made about the returned value when an error is returned. We should not expect it to be `nil`.

Reply:
(for comparison, that is like expecting a map iteration to always go in the same order before they forced randomization, because yes, it did, but no, you should not have relied on it)

Reply:
Maybe it's not in the spec, but it certainly is documented somewhere.